### ISSUE TITLE
Program silently crashes after procesing ~2000 images of 19000 images

### ISSUE THREAD

**Issue by ANTONIOPSD** [Lymphatus/caesium-image-compressor#19]
I tried the v2.0.0-alpha.1 and the program crashes after processing around 2000 images.

![imagen](https://user-images.githubusercontent.com/1978099/80370973-6a1ad400-8891-11ea-9e56-8f743a64a8d3.png)

These are the settings I used:

![imagen](https://user-images.githubusercontent.com/1978099/80371174-c41b9980-8891-11ea-9fd6-c6db26f23af2.png)

**Comment by Lymphatus** [comment]
Does it always crash at the same amount of images? Is the set of images always the same or changing it changes the results too?

**Comment by ANTONIOPSD** [comment]
> 
> 
> Does it always crash at the same amount of images? Is the set of images always the same or changing it changes the results too?

I tested it again and  it's random but always betwen 1500 and 2500. I tested it with the same set of images, I use the "Add folder" option, my images are from my WhatsApp folder that a total of 18GB of content, ( Videos, images, documents...), I'll keep testing.

Also, another thing. Something weird happens, when I select my output folder and start the compressing process, the program first creates a folder named "dia" inside the selected output folder and then, inside that folder all the compressed images and subfolders. I'm using a Windows build in spanish by the way.

This is the origin folder:
![imagen](https://user-images.githubusercontent.com/1978099/80383838-6b093100-88a4-11ea-9441-d4e2f609bedb.png)

This is the output folder:
![imagen](https://user-images.githubusercontent.com/1978099/80383346-bff87780-88a3-11ea-87b7-950835bbb5b9.png)

It creates a folder called "dia" instead of "Media", for some reason it skips the "Me" part.

**Comment by Lymphatus** [comment]
I'm going to test this out for the next release. In the OS X version this does not happen, so should be something with Windows.

**Comment by Lymphatus** [comment]
I pushed a tentative fix for those issues (they are actually 2 different issues) at bb5a8a1d98f1384fbf63655c2068eae103d49f55. They will go out on the alpha-2 release, so you can test it out with your image set.

**Comment by ANTONIOPSD** [comment]
> 
> 
> I pushed a tentative fix for those issues (they are actually 2 different issues) at [bb5a8a1](https://github.com/Lymphatus/caesium-image-compressor/commit/bb5a8a1d98f1384fbf63655c2068eae103d49f55). They will go out on the alpha-2 release, so you can test it out with your image set.

It keeps crashing, but I checked the last processed image (corrupted half processed) and it was a .jpeg (four letters) and I just remembered something about the 1.7 version, in that version everytime I selected a .jpeg image and selected "Input" as format
![imagen](https://user-images.githubusercontent.com/1978099/82161151-9dea8780-989a-11ea-926c-7b929f82d5cc.png)
the preview iimage looked ultra compressed, even at 100% quality

![imagen](https://user-images.githubusercontent.com/1978099/82161190-f02ba880-989a-11ea-9c05-68ffe7b747b0.png)

It also crashed when I added a folder, and started pressing the arrow down key very fast while Auto preview was selected.
(OLD 1.7 VERSION)
![2020-05-18 00-16-40](https://user-images.githubusercontent.com/1978099/82161604-7f868b00-989e-11ea-88af-1eca00fde9bb.gif)

Could the problem in the 1.7 version be related to the new version?

I'm running this on my laptop:

Asus Rog Zephyrus S GX701GWR
Intel Core i7-9750H 6 Core
16GB RAM
Intel 660p M.2 SSD 512GB

**Comment by Lymphatus** [comment]
No, the 2.0 version is a complete rewrite in terms of code. I'm not even looking at the old code right now, because I changed a lot of things under the hood. So bugs in 1.7 are very unlikely affecting this version.

That said, my guess it's that going multicore is causing unexpected crashes at some point. My test set on Windows was a bit smaller than yours (about 6k images) and had problems with 4 threads running, but none with less than that. Your CPU has 12 virtual cores and the alpha.2 version is running on 6 of them (the alpha.1 was running on all 12).

I just compiled a version with just 1 core running. It will be much much slower, but if it runs ok the problem is definitely the multicore part of the code. Please find it below.

And of course thanks for testing it out!
[caesium-image-compressor-2.0.0-alpha.2-win-1.zip](https://github.com/Lymphatus/caesium-image-compressor/files/4642729/caesium-image-compressor-2.0.0-alpha.2-win-1.zip)

**Comment by ANTONIOPSD** [comment]
> 
> 
> No, the 2.0 version is a complete rewrite in terms of code. I'm not even looking at the old code right now, because I changed a lot of things under the hood. So bugs in 1.7 are very unlikely affecting this version.
> 
> That said, my guess it's that going multicore is causing unexpected crashes at some point. My test set on Windows was a bit smaller than yours (about 6k images) and had problems with 4 threads running, but none with less than that. Your CPU has 12 total cores (virtual + physical) and the alpha.2 version is running on 6 of them (the alpha.1 was running on all 12).
> 
> I just compiled a version with just 1 core running. It will be much much slower, but if it runs ok the problem is definitely the multicore part of the code. Please find it below.
> 
> And of course thanks for testing it out!
> [caesium-image-compressor-2.0.0-alpha.2-win-1.zip](https://github.com/Lymphatus/caesium-image-compressor/files/4642729/caesium-image-compressor-2.0.0-alpha.2-win-1.zip)

OK that build worked, but man, I could just disable hyperthreading and some cores in the bios for testing, or change affinity in taskmanager to only one core, there was no need to build a specific version for me 😅, anyway, the problem can be fixed, thanks for your hard work.

**Comment by Lymphatus** [comment]
I can't assume you to know how to disable threads for an application 😁 It took me like minutes, so it's fine. Is there the possibility you could run with 2 ore more threads on the original alpha.2 and see the results? I've only a old Win machine with 2 physical cores right now and the results might be incosistent.
